### ISSUE TITLE
[3.3.5] CC Auras which use their amount to drop not removed by some spells

### ISSUE THREAD

**Issue by Eliminationzx** [TrinityCore/TrinityCore#20234]
**Description:**

CC Auras which use their amount to drop:
- SPELL_AURA_MOD_CONFUSE
- SPELL_AURA_MOD_FEAR
- SPELL_AURA_MOD_STUN
- SPELL_AURA_MOD_ROOT
- SPELL_AURA_TRANSFORM

**Current behaviour:**

CC Auras which use their amount to drop not removed by spell triggered damage.

**Expected behaviour:**

CC Auras which use their amount to drop should be removed by spell triggered damage

**Steps to reproduce the problem:**

1. Cast on target [Psychic Scream](https://woehead.way-of-elendil.fr/?spell=10890).
2. After first step - cast on target these spells : [Mind flay](https://woehead.way-of-elendil.fr/?spell=48156), [Improved Devouring Plague](https://woehead.way-of-elendil.fr/?spell=63627), [ Drain Mana](https://woehead.way-of-elendil.fr/?spell=5138).
3. The result is that fear not removed by these spells.

**Branch(es):** 3.3.5

**TC rev. hash/commit:** https://github.com/TrinityCore/TrinityCore/commit/ccfcf3940e98d333a20684e910483d700933945a

**TDB version:**  TDB 335.63

**Operating system:** Linux debian 8

[Possible bug location.](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/Auras/SpellAuraEffects.cpp#L6159)

**Comment by Jildor** [comment]
Can be caused by this commit?
https://github.com/TrinityCore/TrinityCore/commit/86903f701540fab062c4d7772c201507fc031c63

**Comment by Eliminationzx** [comment]
@Jildor , can you try to revert this changes and test?

**Comment by Jildor** [comment]
Yep, I revert and test and not is the cause :(

**Comment by Jildor** [comment]
Can be related only with channeled spells?

**Comment by Eliminationzx** [comment]
@Jildor , no. It's related with triggered spells and spells without damage like [drain mana](https://woehead.way-of-elendil.fr/?spell=5138).

**Comment by Shauren** [comment]
Why would drain mana break a cc which is supposed to break on damage?

**Comment by Eliminationzx** [comment]
Any news?

**Comment by Eliminationzx** [comment]
Any idea how to fix it?

**Comment by tkrokli** [comment]
Try asking in #trinity (IRC) if you don't see any answer here within the next few days or a week.

**Comment by Killyana** [comment]
> ariel- commented
https://github.com/TrinityCore/TrinityCore/blob/eeb35df7665ee2b462a99faf5950b0a2de366b0c/src/server/game/Spells/Auras/SpellAuraEffects.cpp#L454-L463
Aura amount is calculated at 10% of victim (aura owner in reality) total health, this means damage exceeding this amount will break it.

    We need the formula of chance of breaking.
    Also, fear was thought to be a "dot" and was resisted on "ticks", you don't fear a higher level character and always resist immediately, the fear wears off extremely quick on high-level players and creatures (diminishing returns withstanding).

There is an open PR for that: #20206

**Comment by ariel-** [comment]
1st problem: triggered spells not counting damage on the aura- should be fixed by #21260.
2nd issue is with Drain Mana, it does absolutely no damage to count. I need data from retail.. does it automatically clear CC auras? or depends on drained amount?

**Comment by Shauren** [comment]
There is no mana drain anymore.

**Comment by ariel-** [comment]
And Mana Burn is gone too

**Comment by Eliminationzx** [comment]
@ariel- , thank you for response.
I've found some [information](https://web.archive.org/web/20160121061455/http://wowwiki.wikia.com/wiki/Patch_3.3.3) on drain mana interaction with CC auras.

> [Drain Mana]: Drain Mana now clears away any Crowd Control effects such as Hex, Fear, Polymorph, Blind and Seduce

**Comment by ariel-** [comment]
Mana Burn already has script which handles this:

https://github.com/TrinityCore/TrinityCore/blob/ccfcf3940e98d333a20684e910483d700933945a/src/server/scripts/Spells/spell_priest.cpp#L738-L742

Only missing from Drain Mana, I guess

**Comment by Eliminationzx** [comment]
@ariel-, how about blind, hex, seduction? It will be interrupted?

**Comment by Keader** [comment]
Nope, missing mechanic in that generic script

**Comment by Eliminationzx** [comment]
any progress here?

**Comment by Keader** [comment]
@ariel- , @Eliminationzx Drain mana is wrong.
Should not break C.C.
Check: http://us.battle.net/wow/en/game/patch-notes/3-3-3
Idk where wiki found this information, but is wrong.
Note present in wiki dont even exist.
